openedx/core/djangoapps/course_groups/models.py

```diff
@@ -185,14 +185,35 @@ class CourseCohortsSettings(models.Model):
     @property
     def cohorted_discussions(self):
         """Jsonify the cohorted_discussions"""
-        return json.loads(self._cohorted_discussions)
+
+        segmented_discussions = self.segmented_discussions.all()
```

Comment:
This property "cohorted_discussions" needs to be changed to "divided_discussions", and it will need to include information about HOW each discussion is divided. The UI will need this to display the current state.